mat/triband.go

```diff
@@ -260,6 +260,94 @@ func (t *TriBandDense) Reset() {
 	t.mat.Data = t.mat.Data[:0]
 }
 
+// ReuseAsTriBand changes the receiver to be of size n×n, bandwidth k+1 and of
+// the given kind, re-using the backing data slice if it has sufficient capacity
+// and allocating a new slice otherwise. The backing data is zero on return.
+//
+// The receiver must be empty, n must be positive and k must be non-negative and
+// less than n, otherwise ReuseAsTriBand will panic. To empty the receiver for
+// re-use, Reset should be used.
```

Comment:
There should probably be a warning that changing between U and L will cause a panic if the receiver is not empty.

Reply:
The receiver must be empty and then changing between U/L won't cause a panic.

Reply:
Yes. Sorry.